### PR TITLE
Reduced contribution rate for retirees for long-term care insurance until 2004

### PR DESCRIPTION
**Pull request by MImmesberger**
### What problem do you want to solve?

Fixes the issue discussed in #636.

### CODE REVIEW

**Comment by MImmesberger** [comment]
@hmgaudecker I am aware that the `ges_pflegev_beitr_rente_m_ohne_zusatz_fuer_kinderlose_voller_beitrag` function name is too long. But `ges_pflegev_beitr_rente_m_ohne_zusatz_fuer_kinderlose` should be too long as well (character length > 32) but in #623 tests didn't fail. Do you know why? And any suggestion for a better function name?

**Comment by hmgaudecker** [comment]
Shouldn't matter, after applying `@dates_active` it will only be `ges_pflegev_beitr_rente_m` and that will be checked AFAIAA?

**Comment by MImmesberger** [comment]
The naming convention test fails for a function where I only changed the dates in `@dates_active` (function name: `ges_pflegev_beitr_rente_m_ohne_zusatz_fuer_kinderlose`). I implemented another function with a long name and an alias and there it works. I have no idea why this happens, could you (or someone else) have a look?

**Comment by ChristianZimpelmann** [comment]
This most likely is related to #626.

**Comment by hmgaudecker** [comment]
Ah, good one, thanks!

File: src/_gettsim/social_insurance_contributions/ges_pflegev.py [hunk 1]
```diff
@@ -373,7 +373,35 @@ def ges_pflegev_beitr_selbst_m_zusatz_kinderlos_dummy(
 
 
 @dates_active(
-    start="1995-01-01", end="2004-12-31", change_name="ges_pflegev_beitr_rente_m"
+    start="1995-01-01", end="2004-03-31", change_name="ges_pflegev_beitr_rente_m"
+)
+def ges_pflegev_beitr_rente_m_ohne_zusatz_fuer_kinderlose_reduz_beitrag(
```
[the remark this answers is not in the document]
Comment (hmgaudecker) [reply src/_gettsim/social_insurance_contributions/ges_pflegev.py:378]: Did not send this off...

**Comment by MImmesberger** [comment]
Apparently, the problem persists even after merging the latest changes on main into this branch.

**Comment by hmgaudecker** [comment]
@lars-reimann, the bug occurs because the function is only present for a few months, not an entire year. Hence it is classified as time-independent in in `test_naming_conventions.py` / `time_indep_function_names`, which works on a yearly basis. 

Do you see a simple fix for that, which we could just add to this PR or is it worth its own issue?

**Comment by lars-reimann** [comment]
> @lars-reimann, the bug occurs because the function is only present for a few months, not an entire year. Hence it is classified as time-independent in in `test_naming_conventions.py` / `time_indep_function_names`, which works on a yearly basis.
> 
> Do you see a simple fix for that, which we could just add to this PR or is it worth its own issue?

Maybe you could load all functions using [`load_internal_functions`](https://github.com/iza-institute-of-labor-economics/gettsim/blob/c47684df26ae4c6ae6e7147f44abfa2ead52fbbe/src/_gettsim/functions_loader.py#L145) and then filter them using [`is_time_dependent`](https://github.com/iza-institute-of-labor-economics/gettsim/blob/9f9c7bc8278e75e468335f240d59a14745543aae/src/_gettsim/policy_environment.py#L250).

**Comment by MImmesberger** [comment]
Tests fail because of the error documented in #643. I will fix it in a separate PR.

**Comment by MImmesberger** [comment]
The bug #643 has been fixed. PR is ready for review!

File: src/_gettsim/social_insurance_contributions/ges_pflegev.py [hunk 2]
```diff
@@ -373,9 +373,37 @@ def ges_pflegev_beitr_selbst_m_zusatz_kinderlos_dummy(
...
 @dates_active(
-    start="1995-01-01", end="2004-12-31", change_name="ges_pflegev_beitr_rente_m"
+    start="1995-01-01", end="2004-03-31", change_name="ges_pflegev_beitr_rente_m"
 )
-def ges_pflegev_beitr_rente_m_ohne_zusatz_fuer_kinderlose(
+def ges_pflegev_beitr_rente_m_reduz_beitrag(
+    _ges_krankenv_bemessungsgrundlage_rente_m: float,
+    ges_pflegev_beitr_satz: float,
+) -> float:
+    """Calculating the contribution to health insurance for pension income.
```
Comment (hmgaudecker) [inline src/_gettsim/social_insurance_contributions/ges_pflegev.py:382]:
```suggestion
    """Long-term care insurance contribution from pension income.
```

Please check the others that you touched here. We should get rid of "calculating" and friends everywhere. No need to go through the entire codebase now (bad timing), but if we touch stuff anyhow, it should be fixed.

**Review by hmgaudecker — APPROVED** [review]
Excellent, thanks!

**Pull request merged**